RepeatingRootSource.cc

```diff
@@ -69,7 +69,16 @@ RepeatingRootSource::RepeatingRootSource(std::string const& iName, unsigned int
   for(int i=0; i<nUniqueEvents_; ++i) {
     fillBuffer(i, dataBuffersPerEvent_[i], branches);
     if(eventAuxIndex != -1) {
-      identifierPerEvent_[i] = EventAuxReader(&dataBuffersPerEvent_[i][eventAuxIndex].address_).doWork();
+      auto addr = &dataBuffersPerEvent_[i][eventAuxIndex].address_;
+      EventAuxReader aux_reader;
```

Comment:
Shouldn't this be moved outside of the loop since it is the same for each entry?

Reply:
Are you referring to the default-constructed `aux_reader` or the `addr` variable, which is calculated for each value of `i` (consistent with the previous code)?

Reply:
I was referring to setting of `aux_reader`. I can see now that can't be moved. However, the determination of which `std::function` to create can be moved to outside of the loop.

Reply:
Yep, I'll rework it accordingly.

Reply:
How about just setting a value outside of the loop which says which file type this is and then a `switch` (or ifs) inside the loop that use that to set the `EventAuxReader`? That way we avoid calling`file_->GetKey(...)` repeatedly?

Reply:
I could do that.  I was hoping to do encapsulate the switch logic to something like (e.g.):

```c++
EventAuxReader aux_reader{file_};
for (std::size_t i=0; i != nUniqueEvents_; ++i) {
  auto addr = something_that_depends_on(i);
  identifierPerEvent_[i] = aux_reader.bind_to_branch(addr).doWork();
  ...
}
```
where the constructor of `EventAuxReader` selects the correct function to use.

Reply:
I could see that as well. That should be just as efficient.